aesara/tensor/nnet/sigm.py

```diff
@@ -98,7 +99,7 @@ def c_code_cache_version():
 
 
 # @opt.register_uncanonicalize
-@local_optimizer([sigmoid])
+@local_optimizer(None)
```

Comment:
I am curious if we can keep the original signature? It still applies to sigmoid Ops no?

Reply:
It might only track that `Op` instance.

Reply:
I see. And we can't tell it to track two types of Op?

It's not very important since this is not a default rewrite, I am just curious how it works.

Reply:
We could change the signature to `@local_optimizer([sigmoid, sigmoid_inplace])` so the optimizer is dispatched on the default inplace `Elemwise`. But like @brandonwillard mentioned I don't think that will handle custom `Op`s that use `Sigmoid` (or a subclass of it) as the scalar op.

Reply:
I think the `[sigmoid, sigmoid_inplace]` should be fine. I am not aware of any `Op`s subclassing from `Sigmoid`. Not even these ones do it, right?

Reply:
There are two parts to this: 1) is `tracks` even being used in this case, and 2) we don't want to unnecessarily/unintentionally restrict the applicability of rewrites.

If `tracks` is being used, we will be allowing case 2; if it isn't, then the change is pointless.

The best approach is to fix/update the `tracks` mechanism so that it can operate using `isinstance`/`issubclass` (even for "meta"/"composite" `Op`s like `Elemwise`).

Reply:
I think this is the relevant issue: https://github.com/aesara-devs/aesara/issues/51

Reply:
> whether or not `tracks` even being used in this case

What do you mean? As far as I can see, tracks is working well (at least in the default `canonicalize` / `specialize` rewrites. It also seems to work well for `Elemwise` types.

Reply:
I just created a much more relevant issue for this: https://github.com/aesara-devs/aesara/issues/557

Reply:
> I just created a much more relevant issue for this: #557

Should this PR wait on #557? I'm not clear on which signature (`None` or `[sigmoid, sigmoid_inplace]`) is best at this point.

Reply:
`None` is fine.